spring-boot-actuator/src/main/java/org/springframework/boot/actuate/autoconfigure/HealthIndicatorAutoConfiguration.java

```diff
@@ -276,4 +280,41 @@ public DiskSpaceHealthIndicatorProperties diskSpaceHealthIndicatorProperties() {
 
 	}
 
+	@Configuration
+	@ConditionalOnBean(JavaMailSenderImpl.class)
+	@ConditionalOnProperty(prefix = "management.health.mail", name = "enabled", matchIfMissing = true)
+	public static class MailHealthIndicatorConfiguration {
+
+		@Autowired
+		private HealthAggregator healthAggregator;
+
+		@Autowired(required = false)
+		private Map<String, JavaMailSenderImpl> mailSenders;
+
+		@Bean
+		@ConditionalOnMissingBean(name = "mailSenderHealthIndicator")
+		public HealthIndicator mailHealthIndicator() {
+			if (this.mailSenders.size() == 1) {
+				JavaMailSenderImpl mailSender = this.mailSenders.values().iterator()
+						.next();
+				return createMailSenderHealthIndicator(mailSender);
+			}
+			CompositeHealthIndicator composite = new CompositeHealthIndicator(
+					this.healthAggregator);
+			for (Map.Entry<String, JavaMailSenderImpl> entry : this.mailSenders
+					.entrySet()) {
+				String name = entry.getKey();
+				JavaMailSenderImpl mailSender = entry.getValue();
+				composite.addHealthIndicator(name,
+						createMailSenderHealthIndicator(mailSender));
+			}
+			return composite;
+		}
+
+		private MailHealthIndicator createMailSenderHealthIndicator(
```

Comment:
Looks like a code refactoring that was half applied. should probably be `createMailHealthIndicator`